hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -152,19 +213,67 @@ private String getKeyOrDirectoryName(boolean isDirectory,
   }
 
   @NotNull
+  @SuppressWarnings("parameternumber")
   private Set<String> getDeltaFiles(OmSnapshot fromSnapshot,
-      OmSnapshot toSnapshot, List<String> tablesToLookUp)
-      throws RocksDBException {
-    Set<String> fromSnapshotFiles = RdbUtil.getSSTFilesForComparison(
-        fromSnapshot.getMetadataManager().getStore().getDbLocation().getPath(),
-        tablesToLookUp);
-    Set<String> toSnapshotFiles = RdbUtil.getSSTFilesForComparison(
-        toSnapshot.getMetadataManager().getStore().getDbLocation().getPath(),
-        tablesToLookUp);
+      OmSnapshot toSnapshot, List<String> tablesToLookUp,
+      SnapshotInfo fsInfo, SnapshotInfo tsInfo,
+      String volume, String bucket)
+          throws RocksDBException, IOException {
+    // TODO: Refactor the parameter list
 
     final Set<String> deltaFiles = new HashSet<>();
-    deltaFiles.addAll(fromSnapshotFiles);
-    deltaFiles.addAll(toSnapshotFiles);
+
+    // Check if compaction DAG is available, use that if so
+    if (differ != null && fsInfo != null && tsInfo != null) {
+      // Construct DifferSnapshotInfo
+      final DifferSnapshotInfo fromDSI =
+              getDSIFromSI(fsInfo, fromSnapshot, volume, bucket);
+      final DifferSnapshotInfo toDSI =
+              getDSIFromSI(tsInfo, toSnapshot, volume, bucket);
+
+      LOG.debug("Calling RocksDBCheckpointDiffer");
+      List<String> sstDiffList =
+              differ.getSSTDiffListWithFullPath(toDSI, fromDSI);
+      LOG.debug("SST diff list: {}", sstDiffList);
```

Comment:
nit: This is duplicate. It gets logged in RocksDBCheckpointDiffer: https://github.com/apache/ozone/blob/9fe6d106acc71fa79402880ba6344fcab131185d/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java#L750